src/reducers/MapReducer.js

```diff
@@ -173,7 +191,10 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, mapLoaded: true};
 
         case MAP_REGION_CHANGE:
-            return {...state, lastKnownRegion: payload};
+            const {newRegion} = payload;
+            const sitesFromRegionChange = filterSites({...state, lastKnownRegion: newRegion}, state.filterCriteriaKeys);
+
+            return {...state, lastKnownRegion: newRegion, displaySites: sitesFromRegionChange};
```

Comment:
Now whenever we have a `MAP_REGION_CHANGE` (which happens whenever the map fires off on `regionChangeComplete`) let's run all the sites through the filter function, keeping the filterKeys the same, but checking to see if the sites are on screen.

Reply:
😁